### ISSUE TITLE
polysvp1 uses different, hardcoded value for Tmelt

### ISSUE THREAD

**Issue by PeterCaldwell** [E3SM-Project/scream#254]
p3_functions_math_impl.hpp defines dt=(t - 273.16) and p3_constants.hpp defines Tmelt=273.15. I suspect these values should be identical. Since polysvp1 comes from a published empirical fit, I propose we change Tmelt to 273.16 K. Any objections?

**Comment by ambrad** [comment]
For context, Tmelt comes from physconst.F90:
```
real(r8), public           :: tmelt        = shr_const_tkfrz ! Freezing point of water (K)
```
where
```
   real(R8),parameter :: SHR_CONST_TKFRZ   = 273.15_R8       ! freezing T of fresh water          ~ K
```
and is initialized in micro_p3.F90 through micro_p3_interface.F90's call to micro_p3_utils_init. Perhaps of note,
   real(R8),parameter :: SHR_CONST_TKTRIP  = 273.16_R8       ! triple point of fresh water        ~ K
```
Question: Should polysvp1 be formulated w.r.t. the triple point or the freezing point?

**Comment by PeterCaldwell** [comment]
Good question, @ambrad ! I had assumed the saturation calculation *must* be implemented correctly, but when I actually went back and read the Flatau et al 1992 paper (https://journals.ametsoc.org/doi/pdf/10.1175/1520-0450%281992%29031%3C1507%3APFTSVP%3E2.0.CO%3B2 ) cited as the source of polysvp1, I see that it says to use 273.15 K for the Wexler formulation and 273.16 K for the Goff-Gratch formulation. Polysvp1 is an 8th-order polynomial fit using coefficients provided by the right-hand column of their table 4... and their table 4 is for the *Wexler* implementation. So I think someone made a translation error and we inherited a slightly wrong polysvp from WRF. I just double-checked and E3SM uses a different formulation that seems ok. I also pinged Hugh Morrison because I just can't believe this basic thing would be wrong. I'll issue a bugfix when Hugh or someone else agrees with me that this is wrong.
